extensions/git/src/git.ts

```diff
@@ -1536,6 +1536,25 @@ export class Repository {
 		await this.exec(args);
 	}
 
+	async fastForwardBranch(options: { remote: string; ref: string; readonly cancellationToken?: CancellationToken }): Promise<void> {
+		const args = ['fetch', options.remote, `${options.ref}:${options.ref}`];
+		const spawnOptions: SpawnOptions = {
+			cancellationToken: options.cancellationToken,
+			env: { 'GIT_HTTP_USER_AGENT': this.git.userAgent }
+		};
+
+		try {
+			await this.exec(args, spawnOptions);
+		} catch (err) {
+			if (/! \[rejected\].*\(non-fast-forward\)/m.test(err.stderr || '')) {
+				// The local branch has outgoing changes and it cannot be fast-forwarded.
+				return;
```

Comment:
Should this method throw and we catch the error further up?

Reply:
Fixed!